src/error.rs

```diff
@@ -207,6 +207,15 @@ impl From<HttpClientError> for Error {
     }
 }
 
+impl From<&str> for Error {
```

Comment:
I see where you're coming from here, but I'd prefer to avoid that since this can be easily misused.